### PR TITLE
Add pages to AttendeaseConstants

### PR DESCRIPTION
**Pull request by lucasprag**
### related PRs
- https://github.com/coverall/Attendease/pull/10059
- https://github.com/coverall/Attendease-Third-Party-Themes/pull/2070
- https://github.com/attendease/jekyll-attendease/pull/108

### CODE REVIEW

File: lib/jekyll/attendease_plugin/tags.rb [hunk 1]
```diff
@@ -165,7 +169,12 @@ def render(context)
     privateSite: #{ config['private_site'] },
     authApiEndpoint: "#{ config['auth_host'] }api",
     orgLocales: #{ config['available_portal_locales'] },
-    features: #{ config['features'].to_json }
+    features: #{ config['features'].to_json },
+    pages: #{ pages.to_json },
```
Comment (drlube) [inline lib/jekyll/attendease_plugin/tags.rb]: @lucasprag Does this mean that all pages (published/unpublished and hidden/shown) show up in the page source for the event site?

Comment (lucasprag) [reply lib/jekyll/attendease_plugin/tags.rb]: not quite, I get those pages from `context.registers[:site].data['pages']`
which has only active pages (`event.pages.active`)

**Comment by lucasprag** [comment]
could you review that again please @pgib ? thanks

File: lib/jekyll/attendease_plugin/tags.rb [hunk 2]
```diff
@@ -183,7 +194,9 @@ def render(context)
     orgId: "#{ config['organization_id'] }",
     privateSite: #{ config['private_site'] },
     authApiEndpoint: "#{ config['auth_host'] }api",
-    features: #{ config['features'].to_json }
+    features: #{ config['features'].to_json },
+    pages: #{ pages.to_json },
+    parentPagesAreClickable: #{!!parent_pages_are_clickable}
```
Comment (pgib) [inline lib/jekyll/attendease_plugin/tags.rb]: This should be under settings:

```
settings: { parentPagesAreClickable: #{!!parent_pages_are_clickable} }
```

File: lib/jekyll/attendease_plugin/tags.rb [hunk 3]
```diff
@@ -151,7 +151,18 @@ def initialize(tag_name, url_override, tokens)
...
+        parent_pages_are_clickable = config['parent_pages_are_clickable']
+        site_settings = context.registers[:site].data['site_settings']
+
+        pages = if site_settings['advanced'] && site_settings['advanced']['portal_pages_in_the_event_website']
+                  context.registers[:site].data['portal_pages']
+                else
+                  context.registers[:site].data['pages']
+                end
+
+        pages = pages.reject { |page| page['hidden'] }.sort_by { |page| page['weight'] }
```
Comment (pgib) [inline lib/jekyll/attendease_plugin/tags.rb]: @lucasprag where are you filtering out the page stuff that is not needed for the navigation? (like block instances, and any other attribute that the navigation doesn't use or need) This is what I mean by a minimal data set.

File: lib/jekyll/attendease_plugin/version.rb [hunk 4]
```diff
@@ -1,5 +1,5 @@
 module Jekyll
   module AttendeasePlugin
-    VERSION = '0.6.35'
+    VERSION = '0.6.36.pre1'
```
Comment (pgib) [inline lib/jekyll/attendease_plugin/version.rb]: And the final PR should have this at a stable version.

**Pull request merged**